### PR TITLE
seperate is a typo, the correct word is separate, fixed in a number of places including some function calls

### PR DESCRIPTION
**Pull request by peterennis**
Ref: http://grammarist.com/spelling/separate-vs-seperate/

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

**Comment by qinezh** [comment]
@peterennis Could you sign the contribution license agreement and change a better PR title? Thanks~

**Comment by vwxyzh** [comment]
👍

**Comment by superyyrrzz** [comment]
:+1:

**Comment by peterennis** [comment]
cla signed

On Thursday, July 14, 2016 9:05 AM, Microsoft Contribution License Agreements <msftclas@microsoft.com> wrote:
Hi, 
peterennis signed a Contribution License Agreement for Microsoft, which was approved by the CLA admin - claadmin@microsoft.com.

**Comment by qinezh** [comment]
@peterennis Thanks for your contribution and your PR looks good, but we could not merge your PR before the cla signed  according to the policy...

**Comment by peterennis** [comment]
The message above shows the email I got after the MS cla was signed and approved. I do not know the process or timeline whereby you get the notification. Let me know if I need to do some follow up and where such follow up should take place.

**Comment by qinezh** [comment]
@peterennis Could you rebase the PR since there're some conficts?

**Comment by peterennis** [comment]
Done.
capture45

**Comment by qinezh** [comment]
@peterennis Please update the commit after you rebased...

**Comment by peterennis** [comment]
Updated again.
capture48

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 1]
```diff
@@ -651,14 +651,14 @@ private static void RunBuildSteps(IEnumerable<IDocumentBuildStep> buildSteps, Ac
...
             var k = from fileItem in (
-                    from file in parameters.Files.EnumerateFiles()
-                    from p in (from processor in processors
-                               let priority = processor.GetProcessingPriority(file)
-                               where priority != ProcessingPriority.NotSupported
-                               group processor by priority into ps
-                               orderby ps.Key descending
-                               select ps.ToList()).FirstOrDefault() ?? new List<IDocumentProcessor> { null }
-                    select new { file, p })
+                from file in parameters.Files.EnumerateFiles()
```
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: why removing these space?

Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: why remove these space here?

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 2]
```diff
@@ -675,21 +675,16 @@ private static void RunBuildSteps(IEnumerable<IDocumentBuildStep> buildSteps, Ac
...
+            return from item in toHandleItems.AsParallel().WithDegreeOfParallelism(parameters.MaxParallelism)
                    select new HostService(
                        parameters.Files.DefaultBaseDir,
-                       item == null
-                            ? new FileModel[0]
-                            : from file in item
-                              select Load(processor, parameters.Metadata, parameters.FileMetadata, file.file) into model
-                              where model != null
-                              select model)
+                       from file in item
```
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: Please don't change it in this PR, which only fix typo

File: tools/Deployment/deploy.js [hunk 3]
```diff
@@ -432,12 +436,11 @@ function updateChocoConfigPromiseFn() {
...
 function pushChocoPackage() {
   return function() {
-    return new Promise(function(resolve, reject) {
+    return new Promise(function(reslove, reject) {
       if (!globalOptions.pkgName) {
         reject(new Error('package name can not be null/empty/undefined while pushing choco package'));
       }
-      let promiseFn = execPromiseFn('choco', ['push', globalOptions.pkgName], config.choco.homeDir);
-      promiseFn().then(resolve).catch(reject);
+      return execPromiseFn('choco', ['push', globalOptions.pkgName], config.choco.homeDir)();
```
Comment (qinezh) [inline tools/Deployment/deploy.js]: Seems that you have not resolved the conflicts correctly...

Comment (peterennis) [inline tools/Deployment/deploy.js]: Looks like something from the auto rebase tooling. I will look into it manually.

**Comment by superyyrrzz** [comment]
There are still some unrelated changes. As this PR lasts too long to resolve the conflict, can you start from the latest code then open another PR?

**Comment by peterennis** [comment]
GitHub does not show any problems. Do you really want me to close it and start again?
capture51

**Comment by superyyrrzz** [comment]
@peterennis Yes, it does have no conflicts with dev branch, but you can see some unrelated files changed like `deploy.js`, if you switch to the `Files changed` tab of this PR.

**Comment by peterennis** [comment]
Fixed problems in "Files changed" tab.

**Comment by superyyrrzz** [comment]
Thanks @peterennis for contribution :smile:

**Pull request merged**